include/weak_forms/self_linearizing_forms.h

```diff
@@ -819,6 +864,43 @@ namespace WeakForms
           field_2);
       }
 
+#  else
+
+      template <typename AssemblerScalar_t,
+                std::size_t FieldIndex,
+                typename SymbolicOpField,
+                typename T = EnergyFunctor>
+      auto
+      get_functor_first_derivative(
+        const SymbolicOpField &field,
+        typename std::enable_if<is_sd_functor_op<T>::value>::type * =
+          nullptr) const
+      {
+        static_assert(false,
+                      "The SymEngine library is not enabled in deal.II.");
+        return void;
```

Comment:
Why do you have this code? I have tried to run the tests with only Sacado installed and I hit this `static_assert` in a test. If I remove this code (and the other similar code), the test (weak_forms/energy_functor_01a) passes. I don't understand how this code is instantiated but is this code necessary? I would think that this is only there to get a better error message right? If I tried to call this code without `SymEngine`, I should get an error that the function does not exist, shouldn't I?
Here is the beginning of the error message. Like I said, it's not obvious to me why this code is instantiated:
```
20: ===============================   OUTPUT BEGIN  ===============================                                                                                                                                                          
20: weak_forms/energy_functor_01a.debug: BUILD failed. Output:                                                                                                                                                                               
20: [1/5] Building CXX object tests/weak_forms/CMakeFiles/weak_forms.energy_functor_01a.debug.dir/energy_functor_01a.cc.o                                                                                                                    
20: FAILED: tests/weak_forms/CMakeFiles/weak_forms.energy_functor_01a.debug.dir/energy_functor_01a.cc.o                                                                                                                                      
20: /usr/bin/c++ -DBOOST_NO_AUTO_PTR -DDEBUG -DSOURCE_DIR=\"/home/dealii-weak_forms/tests/weak_forms\" -DTESTING_ENVIRONMENT=light -I/home/dealii-weak_forms/include -I/home/dealii-weak_forms/build/configured_files -I/scratch/source/deali
i/tests -isystem /opt/dealii/include -isystem /opt/dealii/include/deal.II/bundled -isystem /opt/openmpi/include -isystem /opt/boost/include -isystem /opt/trilinos/include -isystem /opt/p4est/FAST/include -pedantic -fPIC -Wall -Wextra -Wm
issing-braces -Woverloaded-virtual -Wpointer-arith -Wsign-compare -Wswitch -Wsynth -Wwrite-strings -Wno-placement-new -Wno-literal-suffix -Wno-psabi -Wno-class-memaccess -fopenmp-simd -std=c++17 -pthread -Wno-parentheses -Wno-unused-loca
l-typedefs -O0 -ggdb -Wa,--compress-debug-sections -MD -MT tests/weak_forms/CMakeFiles/weak_forms.energy_functor_01a.debug.dir/energy_functor_01a.cc.o -MF tests/weak_forms/CMakeFiles/weak_forms.energy_functor_01a.debug.dir/energy_functor
_01a.cc.o.d -o tests/weak_forms/CMakeFiles/weak_forms.energy_functor_01a.debug.dir/energy_functor_01a.cc.o -c /home/dealii-weak_forms/tests/weak_forms/energy_functor_01a.cc                                                                 
20: In file included from /home/dealii-weak_forms/include/weak_forms/weak_forms.h:58,                                                                                                                                                        
20:                  from /home/dealii-weak_forms/tests/weak_forms/energy_functor_01a.cc:23:                                                                                                                                                 
20: /home/dealii-weak_forms/include/weak_forms/self_linearizing_forms.h: In member function 'auto dealiiWeakForms::WeakForms::SelfLinearization::EnergyFunctional<EnergyFunctor>::get_functor_first_derivative(const SymbolicOpField&, typena
me std::enable_if<dealiiWeakForms::WeakForms::is_sd_functor_op<T>::value>::type*) const':                                                                                                                                                    
20: /home/dealii-weak_forms/include/weak_forms/self_linearizing_forms.h:879:23: error: static assertion failed: The SymEngine library is not enabled in deal.II.                                                                             
20:   879 |         static_assert(false,                                                                                                                                                                                                     
20:       |                       ^~~~~                                                                                                                                                                                                      
20: /home/dealii-weak_forms/include/weak_forms/self_linearizing_forms.h:881:16: error: expected primary-expression before 'void'                                                                                                             
20:   881 |         return void;                                                                                                                                                                                                             
20:       |                ^~~~                                                                                                                                                                                                              
20: /home/dealii-weak_forms/include/weak_forms/self_linearizing_forms.h:881:15: error: expected ';' before 'void'                                                                                                                            
20:   881 |         return void;                                                                                                                                                                                                             
20:       |               ^~~~~                                                                                                                                                                                                              
20:       |               ;                                                                                                                                                                                                                  
20: /home/dealii-weak_forms/include/weak_forms/self_linearizing_forms.h:881:16: error: declaration does not declare anything [-fpermissive]                                                                                                  
20:   881 |         return void;                                                                                                                                                                                                             
20:       |                ^~~~ 
```